frontend/src/components/auth/Authy.tsx

```diff
@@ -40,40 +50,79 @@ const Authy = ({
     }
   };
 
+  const handlePasscode = (e: { target: { value: unknown } }) => {
+    const newPasscode = e.target.value as string;
+    if (newPasscode !== null && /^[0-9]{0,6}$/.test(newPasscode)) {
+      setPasscode(newPasscode);
+    }
+  };
+
+  const handleFocus = () => {
+    if (inputRef && inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
   if (authenticatedUser) {
     return <Redirect to={HOME_PAGE} />;
   }
 
+  const boxIndexes = [0, 1, 2, 3, 4, 5];
+
   if (toggle) {
     return (
-      <div style={{ textAlign: "center" }}>
-        <h1>Authy</h1>
-        <form
-          onSubmit={(event) => {
-            event.preventDefault();
-            onAuthySubmit();
-          }}
-        >
-          <div>
-            <p>Enter token from Authy App</p>
-            <input
-              type="text"
-              value={passcode}
-              onChange={(event) => {
-                setPasscode(event.target.value);
-                setError("");
-              }}
-              placeholder="passcode"
-            />
-          </div>
-          <div>
-            <button className="btn btn-primary" type="submit">
-              Verify
-            </button>
-          </div>
-          {error && <p style={{ color: "red" }}>{error}</p>}
-        </form>
-      </div>
+      <>
+        <Box bg="teal.400" height="100vh" onClick={handleFocus}>
+          <Flex bg="white" height="100vh" width="47%" justifyContent="center" alignItems="center">
+            <VStack width="77%" align="flex-start" gap="2vh">
+              <Text variant="login">One last step!</Text>
+              <Text variant="loginSecondary">In order to protect your account, please confirm the authorization code sent to {email}</Text>
+              <Flex direction="row" width="100%" justifyContent="space-around">
+                {boxIndexes.map((boxIndex) => {
+                  return (
+                    <Flex
+                      key={boxIndex}
+                      width="56px"
+                      height="56px"
+                      border="1px solid #A7A7A7"
+                      borderRadius="4px"
+                      justify="center"
+                      align="center"
+                    >
+                      <Text variant="login" textAlign="center">
```

Comment:
I think the text on the buttons should be of weight '500'. When you pull from the login variant it will inherit a weight of 700. Not the biggest deal, i think if you manually specify weight in the component on authy.tsx it will override.